### PR TITLE
Misc. Metropolis fixes

### PR DESCRIPTION
**Pull request by rjwebb**
From https://github.com/canvasxyz/metropolis/issues/60

- [x] Add a "Closed" state for PRs that have been closed. It should be rendered in the same red as "Rejected"
- [x] Rename the current "Github ->" link to "PR ->"
- [x] Add a "Github ->" link that links directly to the FIP file inside the Github repo, for all PRs that have been merged to master.
- [x] Add an "Only" link to each option in the Authors and Status filters, which when clicked toggles off all other checkboxes and filters down to only that option. This should be aligned right and only appear when hovering over an option.
- [ ] Data issues
    - [x] Simple summary not parsed from FIP-0002, FIP-0004
    - [x] FIP-0001: Markdown truncated at the end (lower priority since we're only supposed to show simple summaries)
    - [x] FIP-0013: authors not parsed correctly
    - [ ] FIP-0094: duplicated as a draft - not an issue, the two entries are the "PR" version and the version that is currently in the repository. this distinction is more clear with the new "Closed" state and the new GitHub/PR links
- [x] Authors: *For the purposes of the dropdown filtering only*, try to filter out the author's Github handle and only display that, i.e. "Juan Benet (@jbenet______)" can just become @jbenet______. If the GitHub handle is missing a @ but is an alphanumeric+underscore string, then prepend a @ for the purposes of filtering.
- [x] FIP type parsing: Doesn't need to be perfect but some filtering would be nice
    - [x] FIP-0089: "FRC" should be parsed as FRC
    - [x] FIP-0063: "Core, Networking" can be split into "Core" and "Networking" (then dedupe)
    - [x] Filecoin Common Node Interface V1: "Core - Interface" can be split into "Core", "Interface"
    - [x] FIP-0050: <Technical (Core)> - strip </> so it parses as "Technical (Core)"
- [ ] Sort order: TODO once the Github links have been updated @raykyri.

I've added a "Closed" status to the FIP tracker - if a PR has been merged or closed, then it has this status. This is only applicable to PRs. I wonder if we should also have an "Open" status for open PRs...

The issue with extracting the simple summary for FIP-0002 and FIP-0004 was that the code looks for a Markdown node with the "paragraph" tag, but in these FIPs the simple summary is given as a list.

FIP-0001 was caused by us accessing the "body" of the FIP (i.e. everything after the frontmatter) by splitting the document on "---" and then accessing the section after the frontmatter. This means that if a FIP has "---" in multiple places, it ended up only getting the text between the "---" at the end of the frontmatter and the first "---" in the rest of the article. This has been fixed.

The fix for the authors filter dropdown menu is WIP - so far I have rewritten the `splitAuthors` function to work with more authors strings - it now works more like a real parser, can handle comma separated and space separated fields, distinguishes between names, github usernames and email addresses. This has also fixed the issue with FIP-0013 where authors aren't being displayed correctly.

I've updated the code for handling types and categories to clean up the data we see in the FIP tracker - I found that the set of possible types and categories is closed, so instead of trying to parse the `type` and `category` values in the frontmatter we can just check if the expected values are somewhere in the string.

### CODE REVIEW

**Comment by rjwebb** [comment]
This is ready to merge - it's at a point where the best thing to do next is to merge and for @raykyri to take a look at how the user interface looks as a whole.

File: client/src/pages/dashboard/fip_tracker/fip_entry.tsx [hunk 1]
```diff
@@ -1,39 +1,64 @@
...
-        extractedParagraph = remark().stringify(node)
-        inDesiredSection = false // Stop after finding the first paragraph
-      } else if (node.type === "heading" && node.depth <= 2) {
-        inDesiredSection = false // Stop if another heading of depth 1 or 2 is found
+const FipEntryInner = ({ conversation }: {
+  conversation: FipVersion & {
+    displayed_title: string
+    fip_authors: UserInfo[]
+  }} ) => {
+  console.log(conversation)
```
Comment (raykyri) [inline client/src/pages/dashboard/fip_tracker/fip_entry.tsx:17]: remove?

File: client/src/pages/dashboard/fip_tracker/index.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Button as RadixButton, DropdownMenu, TextField, Select } from "@radix-ui/themes"
+import { Button as RadixButton, DropdownMenu, TextField, Select, Separator } from "@radix-ui/themes"
```
Comment (raykyri) [inline client/src/pages/dashboard/fip_tracker/index.tsx:1]: remove separator

File: client/src/pages/dashboard/fip_tracker/simple_summary.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  let inDesiredSection = false
+  let extractedParagraph: string | null = null
+
+  function visitNode(node) {
+    if (node.type === "heading" && node.children[0].value === title) {
+      inDesiredSection = true
+    } else if (inDesiredSection) {
+      if (node.type === "paragraph" || node.type === "list") {
+        extractedParagraph = remark().stringify(node)
```
Comment (raykyri) [inline client/src/pages/dashboard/fip_tracker/simple_summary.tsx:19]: what if there are multiple paragraphs?

**Comment by raykyri** [comment]
Can you make it so that if someone clicks "only" and only that filter option is selected, then all filter options will be selected?

**Comment by raykyri** [comment]
Can you add the "Only" option for "Status" and "Type"?

**Comment by raykyri** [comment]
Last thing is just to fix `sortFunction` handling of `fip_created` and then this should be good to merge.

**Pull request merged**